pkg/activator/net/revision_backends.go

```diff
@@ -251,17 +255,29 @@ func (rw *revisionWatcher) checkDests(dests dests) {
 	// If we have discovered that this revision cannot be probed directly
 	// do not spend time trying.
 	if rw.podsAddressable {
+		// reprobe set contains the targets that moved from ready to non-ready set.
+		// so they have to be re-probed.
+		reprobe := curDests.becameNonReady(prevDests)
+		if len(reprobe) > 0 {
```

Comment:
If you take the suggestion from below, you might want to ditch this if in favor of a `for ... range`

Reply:
I want to keep the `if`, so I can log only when this happens.